### DISCUSSION TITLE
saving the spacy custom model directly in AWS S3 bucket

### DISCUSSION THREAD

**Discussion by uday-felix** [explosion/spaCy#8826] [Help: Coding & Implementations] [answered]
I was fortunately working on spacy custom model and faced an issue saving the model to S3. Is there a way  could directly save the model into S3 ? 
Spacy only has two methods to save 
nlp.to_disk()
nlp.to_bytes()

Thanks much.

**Comment by polm** [comment, marked as answer]
There is not a function in spaCy to directly save a model to S3. 

Someone recently asked [the same question on StackOverflow](https://stackoverflow.com/questions/68534365/saving-the-custom-model) - the accepted answer there is good. Basically you can use `to_disk` to create a model in a directory, then turn the directory into a zip/tar.gz, then upload that to S3 like any other file.

**Reply by uday-felix** [reply]
Ok. I was looking out for a method which stores directly. Anyway, thank you for the reply.
